BackupBrowser: polish styles and horizontal scrolling

Initsogar opened the pull request:
## Proposed Changes
* Fix focus border being cut when an item is clicked
* Remove `padding-left` on the root node, so we could fit better the backup browser on the space available
* More spacing on `Go Back` button on mobile
* Fix horizontal scrolling when browsing files in high depth
* Improve loading placeholder when opening directories (adding 3-bars instead of just 1)

| Description | Before | After |
|---|---|---|
| Fix focus border being cut when an item is clicked | <img width="197" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/08afd128-f54b-4603-bc43-0c0d3dd1afb2"> | <img width="204" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/21e2d3e9-483c-4493-8663-bfcd4a96f315"> |
| Remove `padding-left` on the root node, so we could fit better the backup browser on the space available | ![image](https://github.com/Automattic/wp-calypso/assets/1488641/3f86b331-3a68-4bc8-9995-8b3436f0c058) | ![image](https://github.com/Automattic/wp-calypso/assets/1488641/543734c4-bfda-4e5e-82a9-796b10d731e6) |
| More spacing on `Go Back` button on mobile | <img width="383" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/a691f53d-f5e6-4e25-9b90-90977efbd991"> | <img width="384" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/9893730b-ef23-41d3-9f5c-712d754f5557"> |
| Fix horizontal scrolling when browsing files in high depth | ![image](https://github.com/Automattic/wp-calypso/assets/1488641/4a741be6-917b-4693-898d-f956aad92342) | ![image](https://github.com/Automattic/wp-calypso/assets/1488641/e3794a4e-7ffa-474c-bee1-e27597f8d515) |
| Improve loading placeholder when opening directories (adding 3-bars instead of just 1) | ![CleanShot 2023-07-13 at 22 58 42](https://github.com/Automattic/wp-calypso/assets/1488641/dcaca10a-1e0e-4f2d-b7e3-30d1f0427794) | ![CleanShot 2023-07-13 at 23 01 17](https://github.com/Automattic/wp-calypso/assets/1488641/eaf0501f-830d-4ccf-b860-25b9458e3dac) |

## Testing Instructions
* Start a Jetpack Cloud or Calypso Live branch
* Select one of your sites with a Jetpack VaultPress Backup plan.
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Navigate through the backup browser and validate the changes proposed:
  * **Fix focus border being cut when an item is clicked**
    * Click on multiple items and ensure the focus indicator is not being cut off.
  * **Remove `padding-left` on the root node, so we could fit better the backup browser on the space available**
    * You should notice the backup browser doesn't have that additional space in the left side
  * **More spacing on `Go Back` button on mobile**
    * Try navigating as a mobile device. The `Go back` button should have more spacing now.
  * **Fix horizontal scrolling when browsing files in high depth**
    * If you have a site with multiple directories nested, you could try to open them all and ensure the horizontal scrollbar is working in the wrapper of the backup browser.
  * **Improve loading placeholder when opening directories (adding 3 items instead of just 1)**
    * When opening a directory, you should see the 3-bars loading indicator.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Initsogar committed:
Improve styles and horizontal scroll

Initsogar committed:
Add 3 items as loading indicators instead of 1

Initsogar committed:
Improve responsive styles

Initsogar committed:
Fix horizontal scroll overlap with last item

grbicsanja commented:
@Initsogar Focus border and loading indicator look great to me 🙌!
As for the spacing:
**padding-left**: It looks like we could expand 4 more pixels to the left to align everything. It would look more tidy.
**go back**: vertical spacing around the go back button still doesn't look good in my browser. It seems to move around based on screen size.
(The image on the left is desktop and on the right it's also desktop but resized to mobile width)
![Slice 6](https://github.com/Automattic/wp-calypso/assets/90194580/e22e0f32-eff5-4eee-95da-2e440c7bef29)

I wasn't able to test the scrolling but should be able to a little later in the day once the new daily backup is up (I added a file with a long name).

Initsogar committed:
Break `go back` link button on 660px breakpoint

Initsogar committed:
Fix margin/padding left alignment

Initsogar commented:
@grbicsanja thanks for the feedback!

> **padding-left:** It looks like we could expand 4 more pixels to the left to align everything. It would look more tidy.
Just adjusted it. The margin-left was necessary to fix the focus indicator, but found a way to adjust it, including the header.

![image](https://github.com/Automattic/wp-calypso/assets/1488641/92fe178e-7622-48d8-931d-fae2dc884280)

> **go back:** vertical spacing around the go back button still doesn't look good in my browser. It seems to move around based on screen size.
It seems we were considering <600px as mobile (which is a good practice), but we are displaying the mobile navigation starting 660px. So just fixed to make it consistent. It is now fixed :)

![image](https://github.com/Automattic/wp-calypso/assets/1488641/40d6176d-cbcc-4dc1-8a96-20cd39b3733f)

Initsogar committed:
Adjust bottom padding on go back link in desktop

grbicsanja reviewed (approved):
LGTM!

jwebbdev reviewed (approved):
LGTM
